opendevin/controller/agent_controller.py

```diff
@@ -174,14 +185,24 @@ def get_agent_state(self):
     async def start_delegate(self, action: AgentDelegateAction):
         AgentCls: Type[Agent] = Agent.get_cls(action.agent)
         agent = AgentCls(llm=self.agent.llm)
+        state = State(
+            inputs=action.inputs or {},
+            iteration=self.state.iteration,
+            max_iterations=self.state.max_iterations,
+            num_of_chars=self.state.num_of_chars,
+            # TODO: the delegate probably doesn't need to know the full history
+            history=self.state.history,
```

Comment:
We share history between delegated agent and parent agent right? And it seems that will broke the check in `delegator_agent` and it will get an AgentFinishAction, not an AgentDelegateObservation. Exception will be thrown. 
```
last_observation = state.history[-1][1]
if not isinstance(last_observation, AgentDelegateObservation):
      raise Exception('Last observation is not an AgentDelegateObservation')
```

Reply:
Good catch! I suspect this was already broken though...?

Aside, @rbren Do you have a roadmap or vision for DelegatorAgent? It seems to make a lot of assumptions about the task itself and I don't know when people should use it.

Reply:
> For example, I think maybe we can call BrowsingAgent to search web in CodeActAgent someday in the future.

Yeah I really wanted to make it happen so I'd like to fix this thing ASAP. And unfortunately we are kind of making a car without a blueprint... lol we are moving fast.